archive/imo/imo1979_q1.lean

```diff
@@ -0,0 +1,288 @@
+/-
+Copyright (c) 2020 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard
+-/
+
+import tactic
+import algebra.big_operators.ring
+import data.nat.parity
+import data.finset.intervals
+import data.padics.padic_norm
+
+/-!
+# IMO 1979 Q1
+
+Let `p` and `q` be positive integers such that
+
+`p/q=1-1/2+1/3-1/4+...-1/1318+1/1319`
+
+Prove that `p` is a multiple of 1979.
+
+## The solution
+
+The proof we formalise is the following. Rewrite the sum as
+`1+1/2+1/3+...+1/1319 - 2*(1/2+1/4+1/6+...+1/1318)=1/660+1/661+...+1/1319
+and now re-arranging as (1/660+1/1319)+(1/661+1/1318)+... we see that
```

Comment:
Bryan I know this might sound ridiculous, but I wrote this part of the code months ago (before a crazy summer of having to port my undergraduate maths course to an online format -- I am only PR'ing now because I found the file last week when I was able to turn my attention back to Lean). As you can probably see, I was actually learning the API for manipulating finite sums at the time (this was not long after the introduction of \sum, I think) and I can quite believe that things have moved on a lot since then. 

What happened was: I learnt about sums as they were back then, knocked off some code, realised I needed to learn about p-adic valuations to finish the job, got bogged down in p-adic numbers, had a bunch of sorries involving valuations on the 1979-adic numbers, then got distracted by real world issues involving actually doing my job, then came back to it last week having realised in the mean time that all I needed to use was the 1979-adic valuation on the rationals. I think you're saying that I should do some serious code review for the sum part, and perhaps have a look at what appeared in mathlib over the summer to see if it is any help. I'll tag this awaiting-author and will try to tidy up these sum lemmas.